core/stringize.h has compile warning on Mac

Via @mjleven  sst-elements issue 542...

/Users/mjleven/sstGIT/split/core/build/install/include/sst/core/stringize.h:154:5: warning: control reaches end of
non-void function [-Wreturn-type]
}

Looks like operator() needs to return something.

    struct escaped_list_separator {
    typedef std::string::const_iterator iter;
    std::string e;
    std::string q;
    std::string s;

    escaped_list_separator(const std::string &esc="\\", const std::string &sep=",", const std::string &quote="\"")
        : e(esc), s(sep), q(quote) { }


    /**
     * @return pair<iter, iter> = <tok_end, next_tok>
     */
    bool operator()(iter &first, iter last, std::string &token) {
    token.clear();

    bool inside_quotes = false;
    bool in_escape = false;
    while ( first != last ) {
        char c = *first++;

        if ( in_escape ) {
            token += c;
            in_escape = false;
        } else if ( s.find(c) != std::string::npos && !inside_quotes ) {
            /* Separator found */
            break;
        } else if ( q.find(c) != std::string::npos ) {
            inside_quotes = !inside_quotes;
        } else if ( e.find(c) != std::string::npos ) {
            in_escape = true;
        } else {
            token += c;
        }
    }
    }

    };

--- a/src/sst/core/stringize.h
+++ b/src/sst/core/stringize.h
@@ -129,7 +129,7 @@ struct escaped_list_separator {
     /**
      * @return pair<iter, iter> = <tok_end, next_tok>
      */
-    bool operator()(iter &first, iter last, std::string &token) {
+    void operator()(iter &first, iter last, std::string &token) {
         token.clear();
 
         bool inside_quotes = false;

